src/Components/Ignitor/src/CancellableOperation.cs

```diff
@@ -14,7 +14,7 @@ public CancellableOperation(TimeSpan? timeout, CancellationToken cancellationTok
         {
             Timeout = timeout;
 
-            Completion = new TaskCompletionSource<TResult>(TaskContinuationOptions.RunContinuationsAsynchronously);
+            Completion = new TaskCompletionSource<TResult>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Just want to be clear that (if this is the same as with the JSInterop issue we discussed before), this is a change of behavior, not some neutral refactoring.

Since it's in test code only, I'm not deeply concerned about working out every possible edge case with usage - we know we're the only consumer of this code. We know that Blazor Server is basically async on all its interactions so there's no reason to fear this change is weakening the guarantees enforced by the tests, so I'm fine with this as is.

I'm raising it only in case @pranavkm or anyone else can anticipate any problems caused by changing this behavior to be truly async.

Reply:
While it's certainly a behavioral change, the fixed code is certainly doing what the author intended and the old code is not. I'd be really worried if a continuation relied on running inline in a TCS.SetResult() call in any case since it's such a subtle behavior. If we really want to run code inline, we should use a callback, not a TCS.

Reply:
> I'd be really worried if a continuation relied on running inline in a TCS.SetResult() call 

We had a bunch of tests that relied on this exact behavior from product code. I ran the e2e tests locally a couple of times with this change and they all passed so I feel pretty confident about this change. If we have tests that rely on this behavior, we can always correct this behavior.